src/reason_pprint_ast.ml

```diff
@@ -1296,303 +1335,488 @@ let convertIsListyToIsSequencey isListyImpl =
 
-     This doesn't work well because we don't have special interaction with
-     lists, that ensure those kinds of end-of-line comments cause that list to
-     *always* breaks. Currently, the item compressing onto one line
-     changes the meaning entirely. In the following case, the comment becomes
-     attached to the entire let binding. That is a source of non-idempotent
-     formatting.
+(** [getLocFromLayout] recursively takes the unioned location of its children,
+ *  and returns the max one
+ *)
+let rec getLocFromLayout = function
+  | Sequence (listConfig, subLayouts) ->
+     let locs = List.map getLocFromLayout subLayouts in
+     List.fold_left unionLoc None locs
+  | Label (formatter, left, right) ->
+     let leftLoc = getLocFromLayout left in
+     let rightLoc = getLocFromLayout right in
+     unionLoc leftLoc rightLoc
+  | SourceMap (loc, _) ->
+     Some loc
+  | WithComment (_, sub) ->
+     getLocFromLayout sub
+  | _ -> None
 
-       let x = {item:blah}  /* On the entire let binding */
-       and y = {
-       };
+(**
+ * Returns true if loc1 contains loc2
+ *)
+let containLoc loc1 loc2 =
+  loc1.loc_start.Lexing.pos_cnum <= loc2.loc_start.Lexing.pos_cnum &&
+  loc1.loc_end.Lexing.pos_cnum >= loc2.loc_end.Lexing.pos_cnum
 
-   *)
-  let (endOfLineComments, unconsumed) = extractComments afterStart (
-    fun comAttachFirst comAttachLast comPhysFirst comPhysLast ->
-      let entirelyPhysicallyAfter = comPhysFirst >= lastChar && comPhysLast >= lastChar in
-      (* In case we're converting from OCaml, OCaml's parser parses | X with
-         the location of X but only sometimes. Reason's parser always parses
-         the X pattern as having the location start of |.
-         However, in either case the attachment location is always at the
-         location of the bar.
-         This - 2 subtraction hack is only to pretend that OCaml had located
-         the X pattern consistent with Reason.
-      *)
-      let firstCharPossiblyBar = (comAttachFirst == firstChar - 2) in
-      let attachmentEnclosesEntireThing = comAttachFirst == firstChar || firstCharPossiblyBar in
-      attachmentEnclosesEntireThing && entirelyPhysicallyAfter
-  ) in
-  (onItem, endOfLineComments, unconsumed)
-
-(** Only finds the comments that occurred before the location.  When you don't
-    have a wrapping list to interleave comments in, this is the least error
-    prone function to use.
+(**
+ * Returns true if loc1 is before loc2
  *)
-and partitionPreviousItemComments loc comments =
-  let firstChar = loc.loc_start.Lexing.pos_cnum in
-  extractComments comments (
-    fun comAttachFirst comAttachLast comPhysFirst comPhysLast ->
-      let entirelyPhysicallyAbove = comPhysFirst < firstChar && comPhysLast < firstChar in
-      entirelyPhysicallyAbove
-  )
+let beforeLoc loc1 loc2 =
+  loc1.loc_end.Lexing.pos_cnum <= loc2.loc_start.Lexing.pos_cnum
 
+let rec attachEOLComment layout txt =
+  WithComment (txt, layout)
```

Comment:
I'm trying to find all the uses of `WithComment`, and I'm only able to find ones that model end of line comments. Should `WithComment` be called `WithEndOfLineComment`?